R/data_tabulate.R

```diff
@@ -465,6 +483,87 @@ as.data.frame.datawizard_tables <- function(x,
 as.data.frame.datawizard_crosstabs <- as.data.frame.datawizard_tables
 
 
+# as.table --------------------
+
+#' @rdname data_tabulate
+#' @export
+as.table.datawizard_table <- function(x, remove_na = TRUE, ...) {
+  # sanity check - the data frame method returns a list
+  if (!is.data.frame(x)) {
+    x <- x[[1]]
+  }
+  if (remove_na) {
+    # remove NA values from the table
+    x <- x[!is.na(x$Value), ]
+  }
+  as.table(stats::setNames(x[["N"]], x$Value))
+}
+
+#' @export
+as.table.datawizard_tables <- function(x, remove_na = TRUE, ...) {
+  out <- lapply(x, as.table.datawizard_table, remove_na = remove_na, ...)
+  # if only one table is returned, "unlist"
+  if (length(out) == 1) {
+    out <- out[[1]]
+  }
+  out
+}
+
+#' @export
+as.table.datawizard_crosstab <- function(x, remove_na = TRUE, ...) {
+  # data frame is in a list, so we extract the first element
+  if (is.data.frame(x)) {
+    out <- x
+  } else {
+    out <- x[[1]]
+  }
```

Comment:
There's a similar but slightly different logic at the top of `as.table.datawizard_table()`, can you harmonize them?